sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q9/simplified.txt

```diff
@@ -2,77 +2,72 @@ WholeStageCodegen (1)
+                    Filter [ss_quantity]
+                      ColumnarToRow
+                        InputAdapter
+                          Scan parquet spark_catalog.default.store_sales [ss_quantity,ss_ext_discount_amt,ss_net_paid,ss_sold_date_sk]
     ReusedSubquery [mergedValue] #1
     ReusedSubquery [mergedValue] #1
     Subquery #2
       WholeStageCodegen (2)
-        Project [count(1),avg(ss_ext_discount_amt),avg(ss_net_paid)]
-          HashAggregate [count,sum,count,sum,count] [count(1),avg(UnscaledValue(ss_ext_discount_amt)),avg(UnscaledValue(ss_net_paid)),count(1),avg(ss_ext_discount_amt),avg(ss_net_paid),count,sum,count,sum,count]
-            InputAdapter
-              Exchange #2
-                WholeStageCodegen (1)
-                  HashAggregate [ss_ext_discount_amt,ss_net_paid] [count,sum,count,sum,count,count,sum,count,sum,count]
-                    Project [ss_ext_discount_amt,ss_net_paid]
-                      Filter [ss_quantity]
-                        ColumnarToRow
-                          InputAdapter
-                            Scan parquet spark_catalog.default.store_sales [ss_quantity,ss_ext_discount_amt,ss_net_paid,ss_sold_date_sk]
+        HashAggregate [count,sum,count,sum,count] [count(1),avg(UnscaledValue(ss_ext_discount_amt)),avg(UnscaledValue(ss_net_paid)),mergedValue,count,sum,count,sum,count]
+          InputAdapter
+            Exchange #2
+              WholeStageCodegen (1)
+                HashAggregate [ss_ext_discount_amt,ss_net_paid] [count,sum,count,sum,count,count,sum,count,sum,count]
+                  Project [ss_ext_discount_amt,ss_net_paid]
+                    Filter [ss_quantity]
+                      ColumnarToRow
+                        InputAdapter
+                          Scan parquet spark_catalog.default.store_sales [ss_quantity,ss_ext_discount_amt,ss_net_paid,ss_sold_date_sk]
     ReusedSubquery [mergedValue] #2
     ReusedSubquery [mergedValue] #2
     Subquery #3
       WholeStageCodegen (2)
-        Project [count(1),avg(ss_ext_discount_amt),avg(ss_net_paid)]
-          HashAggregate [count,sum,count,sum,count] [count(1),avg(UnscaledValue(ss_ext_discount_amt)),avg(UnscaledValue(ss_net_paid)),count(1),avg(ss_ext_discount_amt),avg(ss_net_paid),count,sum,count,sum,count]
-            InputAdapter
-              Exchange #3
-                WholeStageCodegen (1)
-                  HashAggregate [ss_ext_discount_amt,ss_net_paid] [count,sum,count,sum,count,count,sum,count,sum,count]
-                    Project [ss_ext_discount_amt,ss_net_paid]
-                      Filter [ss_quantity]
-                        ColumnarToRow
-                          InputAdapter
-                            Scan parquet spark_catalog.default.store_sales [ss_quantity,ss_ext_discount_amt,ss_net_paid,ss_sold_date_sk]
+        HashAggregate [count,sum,count,sum,count] [count(1),avg(UnscaledValue(ss_ext_discount_amt)),avg(UnscaledValue(ss_net_paid)),mergedValue,count,sum,count,sum,count]
```

Comment:
I don't understand why there is a change here. Is it because of CollapseProject? Is this correct? @cloud-fan